diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -58,6 +59,101 @@ int DifferentialFunctionComparator::cmpGEPs(const GEPOperator *GEPL,
+        && isa<ArrayType>(GEPR->getSourceElementType())) {
+        // The number of elements of two enums can be different and
+        // its not considered as semantic change
+        if (GEPL->getSourceElementType()->getArrayNumElements()
+            == GEPR->getSourceElementType()->getArrayNumElements())
+            return OriginalResult;
+
+        if (int Res =
+                    cmpNumbers(GEPL->getNumOperands(), GEPR->getNumOperands()))
+            return Res;
+
+        for (unsigned i = 0, e = GEPL->getNumOperands(); i != e; ++i) {
+            if (int Res = cmpValues(GEPL->getOperand(i), GEPR->getOperand(i)))
+                return Res;
+        }
+        if (GEPL->hasAllConstantIndices() || GEPR->hasAllConstantIndices())
+            return OriginalResult;
+
+        auto instrL = dyn_cast<Instruction>(GEPL->getOperand(2));
+        auto instrR = dyn_cast<Instruction>(GEPR->getOperand(2));
+
+        if (instrL && !isa<ZExtInst>(instrL) && !isa<SExtInst>(instrL)
+            && !isa<TruncInst>(instrL))
+            return OriginalResult;
+        if (instrR && !isa<ZExtInst>(instrR) && !isa<SExtInst>(instrR)
+            && !isa<TruncInst>(instrR))
+            return OriginalResult;
+
+        auto indexL = instrL ? instrL->getOperand(0) : GEPL->getOperand(2);
+        auto indexR = instrR ? instrR->getOperand(0) : GEPR->getOperand(2);
+
+        if (!indexL->getType()->isIntegerTy()
+            || !indexR->getType()->isIntegerTy())
+            return OriginalResult;
+
+        const GetElementPtrInst *GEPiL = dyn_cast<GetElementPtrInst>(GEPL);
+        const GetElementPtrInst *GEPiR = dyn_cast<GetElementPtrInst>(GEPR);
+        BasicBlock::const_iterator InstL = GEPiL->getParent()->begin(),
+                                   InstLE = GEPiL->getParent()->end();
+        BasicBlock::const_iterator InstR = GEPiR->getParent()->begin(),
+                                   InstRE = GEPiR->getParent()->end();
+
+        while (InstL != InstLE && InstR != InstRE) {
+            auto dbgInstL = dyn_cast<DbgValueInst>(InstL);
+            auto dbgInstR = dyn_cast<DbgValueInst>(InstR);
+
+            if (!dbgInstL) {
+                InstL++;
+                continue;
+            }
+            if (!dbgInstR) {
+                InstR++;
+                continue;
+            }
+
+            auto ValMDL = dyn_cast<MetadataAsValue>(dbgInstL->getOperand(0))
+                                  ->getMetadata();
+            auto dbgValL = dyn_cast<ValueAsMetadata>(ValMDL)->getValue();
+
+            auto ValMDR = dyn_cast<MetadataAsValue>(dbgInstR->getOperand(0))
+                                  ->getMetadata();
+            auto dbgValR = dyn_cast<ValueAsMetadata>(ValMDR)->getValue();
+
+            if (dbgValL != indexL) {
+                InstL++;
+                continue;
+            }
+            if (dbgValR != indexR) {
+                InstR++;
+                continue;
+            }
+            break;
+        }
```

Comment:
This is a weird loop, I'd prefer to have a separate function that gets a value and returns its metadata, or even its type. It could be called e.g. `getVariableMetadata` and it should be located in `DebugInfo`.

Reply:
Implemented `getVariableTypeInfo` function in `DebugInfo`, which returns the type of given value.